localstack/services/awslambda/lambda_api.py

```diff
@@ -515,7 +517,21 @@ def create_function():
         lambda_arn_to_runtime[arn] = data['Runtime']
         lambda_arn_to_envvars[arn] = data.get('Environment', {}).get('Variables', {})
         result = set_function_code(data['Code'], lambda_name)
-        return result or jsonify({})
+        result.update({
+            "DeadLetterConfig": data['DeadLetterConfig'],
```

Comment:
This should be `data.get('DeadLetterConfig')`, as `DeadLetterConfig` is optional.